core/src/main/java/org/infinispan/remoting/responses/ClusteredGetResponseValidityFilter.java

```diff
@@ -33,8 +33,8 @@ public ClusteredGetResponseValidityFilter(Collection<Address> targets, Address s
    public boolean isAcceptable(Response response, Address address) {
       if (targets.contains(address)) {
          missingResponses--;
-         if (response instanceof SuccessfulResponse) {
-            validResponses++;
+         if (response instanceof SuccessfulResponse || response instanceof ExceptionResponse) {
```

Comment:
@pruivo @tristantarrant @galderz @wburns Hi, so we have a stalemate with @danberindei we'd like to decide:

When a remote get is executed (to say, 2 owners), and first owner responds with an exception, there are two things we can do:
1. ignore the exception (swallow it) and wait for the second response (possibly without exception) - that's what master does
2. report the exception immediately (suggested change)

An exception can be thrown from two reasons:
a) bug in Infinispan code
b) AvailabilityException or similar

Note: AE can be thrown on one owner and not on another under certain timing; it should not happen in steady state

Pros:
- fail fast, both during development (for a)-style exceptions), and for b) as the user code is likely to hit AE more often (so the code will deal with it properly). Swallowing exceptions hides problems.
- when all owners would throw an exception, we won't get just RpcException without any other info (until this PR, it would end up with timeout, other fixes in the PR at least throw RpcException when there's no acceptable response)
- don't do another RPC when the first one already failed - just hoping for the best when we're likely to get the same response

Cons:
- the value can be available, and could be of use to the user (rather than the exception)
- it is a change of behaviour (in a situation when the exception can happen with different timing, though)